src/dos/cdrom_image.cpp

```diff
@@ -111,11 +120,12 @@ bool CDROM_Interface_Image::BinaryFile::seek(Bit32u offset)
 Bit32u CDROM_Interface_Image::BinaryFile::decode(Bit16s *buffer, Bit32u desired_track_frames)
 {
 	// Guard: only proceed with a valid file
-	if (file == nullptr) return 0;
+	if (file == nullptr)
+		return 0;
 
 	file->read((char*)buffer, desired_track_frames * BYTES_PER_REDBOOK_PCM_FRAME);
-	return static_cast<Bit32u>(ceil(
-		static_cast<float>(file->gcount()) / BYTES_PER_REDBOOK_PCM_FRAME));
+	const uint32_t bytes_read = file->gcount();
+	return (bytes_read + BYTES_PER_REDBOOK_PCM_FRAME - 1) / BYTES_PER_REDBOOK_PCM_FRAME;
```

Comment:
Maybe I'm being pedantic here, as it'll unlikely and cd image will be larger than 2GB, but perhaps a warning log should appear in here, or maybe interface change is required?

`gcount()` returns [`std::streamsize`](https://en.cppreference.com/w/cpp/io/streamsize); on 64-bit archs it'll be signed 64-bit wide integer, so there's a possibility of overflow here.

Reply:
The more type-correct the better, and the less trouble we'll get in.
I was using that 32bit `bytes_read` for two purposes; better name the value being used, but more importantly to "pre-cast" that integer math below to 32bit to avoid casting down from 64 bits to satisfy the 32bit return value.
Now that I've gone with 64 (as it should be), I've similarly bumped the return types to 64 and no need for casting either.